backends/ebpf/run-ebpf-test.py

```diff
@@ -14,30 +14,61 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-# Runs the p4c-ebpf compiler on a P4-16 program
-# and then runs the program on specified input packets
+""" Contains different eBPF models and specifies their individual behavior
+    Currently five phases are defined:
+   1. Invokes the specified compiler on a provided p4 file.
+   2. Parses an stf file and generates an pcap output.
+   3. Loads the generated template or compiles it to a runnable binary.
+   4. Feeds the generated pcap test packets into the P4 "filter"
+   5. Evaluates the output with the expected result from the .stf file
+"""
 
 from __future__ import print_function
 import sys
 import os
 import stat
 import tempfile
 import shutil
-from ebpftargets import EBPFFactory
-sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)) + '/../../tools')
+sys.path.insert(0, os.path.dirname(
+    os.path.abspath(__file__)) + '/../../tools')
 from testutils import *
 
 
+def import_from(module, name):
+    """ Try to import a module and class directly instead of the typical
+        Python method. Allows for dynamic imports. """
+    module = __import__(module, fromlist=[name])
+    return getattr(module, name)
+
+
+class EBPFFactory(object):
+    """ Generator class.
+     Returns a target subclass based on the provided target option."""
+    @staticmethod
+    def create(tmpdir, options, template, outputs):
+        target_name = "targets." + options.target + "_target"
+        target_class = "Target"
+        report_output(outputs["stdout"], options.verbose,
+                      "Loading target ", target_name)
+        try:
+            EBPFTarget = import_from(target_name, target_class)
```

Comment:
This is quite nice since it allows for dynamic imports. Depending on the specified target (command line), we import a different python class. When running xdp for example, we just need to specify --target xdp and this fragment will import targets/xdp_target.py. If the file does not exist, it throws an error.